src/Microsoft.SqlTools.ServiceLayer/Connection/ConnectionInfo.cs

```diff
@@ -60,5 +65,84 @@ public ConnectionInfo(ISqlConnectionFactory factory, string ownerUri, Connection
         /// Returns true is the db connection is to a SQL db
         /// </summary>
         public bool IsAzure { get; set; }
+
+        /// <summary>
+        /// Try to get the DbConnection associated with the given connection type string. 
+        /// </summary>
+        public bool TryGetConnection(string connectionType, out DbConnection connection)
+        {
+            if (connectionType == null)
+            {
+                connectionType = ConnectionType.Default;
+            }
+
+            return ConnectionTypeToConnectionMap.TryGetValue(connectionType, out connection);
+        }
+
+        /// <summary>
+        /// Get a List of all DbConnection instances held by this ConnectionInfo
+        /// </summary>
+        public List<DbConnection> GetAllConnections()
+        {
+            return ConnectionTypeToConnectionMap.Values.ToList();
+        }
+
+        /// <summary>
+        /// Get a list of all connection type strings held by this ConnectionInfo
+        /// </summary>
+        /// <returns></returns>
+        public List<string> GetAllConnectionTypes()
+        {
+            return ConnectionTypeToConnectionMap.Keys.ToList();
+        }
+
+        /// <summary>
+        /// Gets the count of DbConnectioninstances held by this ConnectionInfo 
+        /// </summary>
+        public int CountConnections()
+        {
+            return ConnectionTypeToConnectionMap.Count;
+        }
+
+        /// <summary>
+        /// Adds a DbConnection to this object and associates it with the given 
+        /// connection type string. If a connection already exists with an identical 
+        /// connection type string, it is overwritten. Ignores calls where connectionType = null
+        /// </summary>
+        public void AddConnection(string connectionType, DbConnection connection)
+        {
+            if (connectionType != null)
```

Comment:
These checks for null suggest to me that we should definitely be using an Enum here.
Alternatively, we should consider using the Utils.Validate* methods (or whatever they're called) to check for null/whitespace and throw on those values.